app/src/org/commcare/android/view/GridEntityView.java

```diff
@@ -336,89 +349,112 @@ public void setViews(Context context, Detail detail, Entity entity){
 		View retVal;
-		if(multimediaType.equals(EntityView.FORM_IMAGE)){
-			retVal = new ImageView(context);
-	        if(horzAlign.equals("center")) {
-	            ((ImageView)retVal).setScaleType(ScaleType.CENTER_INSIDE);
-	        } else if(horzAlign.equals("left")) {
-	            ((ImageView)retVal).setScaleType(ScaleType.FIT_START);
-	        } else if(horzAlign.equals("right")) {
-	            ((ImageView)retVal).setScaleType(ScaleType.FIT_END);
-	        }  
-			retVal.setPadding(CELL_PADDING_HORIZONTAL,CELL_PADDING_VERTICAL,CELL_PADDING_HORIZONTAL,CELL_PADDING_VERTICAL);
-			// image loading is handled asyncronously by the TCImageLoader class to allow smooth scrolling
-			if(rowData != null && !rowData.equals("")){
-			    if(mImageLoader != null){
-			        mImageLoader.display(rowData, ((ImageView)retVal), R.drawable.info_bubble);
-			    } else{
-			        Bitmap b = ViewUtil.inflateDisplayImage(getContext(), rowData);
-			        ((ImageView)retVal).setImageBitmap(b);
-			    }
-			}
-		} 
-		else if(multimediaType.equals(EntityView.FORM_AUDIO)){
-    		if (rowData != null & rowData.length() > 0) {
-    			retVal = new AudioButton(context, rowData, uniqueId, controller, true);
-    		}
-    		else {
-    			retVal = new AudioButton(context, rowData, uniqueId, controller, false);
-    		}
-		} else{
-			retVal = new TextView(context);
-			
-			//the html spanner currently does weird stuff like converts "a  a" into "a a"
-			//so we've gotta mirror that for the search text. Booooo. I dunno if there's any
-			//other other side effects (newlines? nbsp?)
-			
-			String htmlIfiedSearchField = searchField == null ? searchField : MarkupUtil.getSpannable(searchField).toString();
-			
-			if(cssid !=null && !cssid.equals("none")){
-			    // user defined a style we want to use
-			    Spannable mSpannable = MarkupUtil.getCustomSpannable(cssid, rowData);
-			    EntityView.highlightSearches(this.getContext(), searchTerms, mSpannable, htmlIfiedSearchField, mFuzzySearchEnabled, mIsAsynchronous);
-			    ((TextView)retVal).setText(mSpannable);
-			} else{
-			    // just process inline markup 
-			    Spannable mSpannable = MarkupUtil.returnCSS(rowData);
-			    EntityView.highlightSearches(this.getContext(), searchTerms, mSpannable, htmlIfiedSearchField, mFuzzySearchEnabled, mIsAsynchronous);
-			    ((TextView)retVal).setText(mSpannable);
-			}
+        switch (multimediaType) {
+            case EntityView.FORM_IMAGE:
+                retVal = new ImageView(context);
+                if (horzAlign.equals("center")) {
+                    ((ImageView)retVal).setScaleType(ScaleType.CENTER_INSIDE);
+                } else if (horzAlign.equals("left")) {
+                    ((ImageView)retVal).setScaleType(ScaleType.FIT_START);
+                } else if (horzAlign.equals("right")) {
+                    ((ImageView)retVal).setScaleType(ScaleType.FIT_END);
+                }
```

Comment:
switch statements have such stupid semantics in Java. Wish there was plain old pattern matching... like Scala http://docs.scala-lang.org/tutorials/tour/pattern-matching.html